autocomplete.js

```diff
@@ -5,12 +5,17 @@ function createAwsAutocompleteFunction(
   methodName,
   outputDataPath,
   [valuePath, labelPath] = [],
+  payloadBuilder = () => ({}),
 ) {
   return async (query, params, codeCommitClient) => {
-    const fetchResult = await fetchRecursively(codeCommitClient, {
-      methodName,
-      outputDataPath,
-    }).catch((error) => {
+    const fetchResult = await fetchRecursively(
+      codeCommitClient,
+      {
+        methodName,
+        outputDataPath,
+      },
+      payloadBuilder(params),
```

Comment:
There's no need to call a function just to obtain an empty objec. It would be more explicit to check if the function is passed. If it is, then to call it, otherwise to pass default payload.
If you need a default parameter for the function just use `null` then.